Documentation/kubectl-tips-and-tricks.md

```diff
@@ -0,0 +1,72 @@
+# Kubernetes kubectl Tips and Tricks
+
+This is a collection of tips and tricks for the [kubectl](https://coreos.com/kubernetes/docs/latest/configure-kubectl.html) command line tool. All of these tricks assume you have a working Kubernetes cluster and kubectl tool.
+
+## Integrating Kubernetes with the Fabric Tool
+
+Kubernetes has a database of nodes in the cluster which can be queried with `kubectl get nodes`. This is a powerful database for automation and integration with existing tools. One powerful tool is the [Fabric SSH utility](http://www.fabfile.org/) which is known as a `fabfile.py` in some circles.
+
+Using the a pre-prepared [Kubernetes fabfile.py](https://github.com/coreos/coreos-kubernetes/blob/master/Documentation/fabric/fabfile.py) you can integrate Kubernetes Nodes and labels with fabric. Here is an example session using this integration:
+
+```
+$ kubectl label node ip-10-0-0-50.us-west-2.compute.internal my-special-label=true
+$ fab -u core -R my-special-label=true -- date
+[52.26.54.211] Executing task '<remainder>'
+[52.26.54.211] run: date
+[52.26.54.211] out: Thu Feb 16 06:54:37 UTC 2017
+[52.26.54.211] out:
+```
+
+By default this will use the `ExternalIP` of nodes. If you want to change this to say the `InternalIP` export the environment variable `FAB_KUBE_NODE_ADDRESS_TYPE=InternalIP`.
+
+![Demo of fabric integration](http://i.imgur.com/YYHmvMl.gif)
+
+## Pod Operations with kubectl
+
+### Filter Kubernetes Pods by Time
+
+The `jq` tool is a lightweight JSON processor that can do comparisons. By combining `jq` with `kubectl` JSON output you can make complex queries like filtering all resources by their create date.
```

Comment:
You can do most of this with -o jsonpath=

For example: 
$ kubectl get pods --all-namespaces --sort-by='.metadata.creationTimestamp' -o jsonpath='{range .items[*]}{.metadata.name}, {.metadata.creationTimestamp}{"\n"}{end}'
kube-controller-manager-3993611301-1606x, 2017-02-15T19:09:29Z
kube-proxy-n3tt3, 2017-02-15T19:09:29Z
kube-proxy-lh3w8, 2017-02-15T19:09:29Z
kube-proxy-k8tlw, 2017-02-15T19:09:29Z
kube-flannel-jcwjz, 2017-02-15T19:09:29Z
kube-flannel-gc742, 2017-02-15T19:09:29Z
kube-flannel-g55w7, 2017-02-15T19:09:29Z
kube-flannel-d274g, 2017-02-15T19:09:29Z
kube-flannel-9rpck, 2017-02-15T19:09:29Z
kube-flannel-16z8c, 2017-02-15T19:09:29Z
kube-proxy-v57zp, 2017-02-15T19:09:29Z
kube-controller-manager-3993611301-l4ztc, 2017-02-15T19:09:29Z
kube-scheduler-850181384-f7p29, 2017-02-15T19:09:29Z
checkpoint-installer-6sbvb, 2017-02-15T19:09:29Z
kube-scheduler-850181384-1cj5w, 2017-02-15T19:09:29Z
kube-apiserver-jg45g, 2017-02-15T19:09:29Z
kube-proxy-qj030, 2017-02-15T19:09:29Z
kube-proxy-nm0j2, 2017-02-15T19:09:29Z
pod-checkpointer-ip-10-0-0-50.us-west-1.compute.internal, 2017-02-15T19:10:47Z
prometheus-2787677268-2lrbh, 2017-02-15T19:11:31Z
prometheus-node-exporter-5m5pj, 2017-02-15T19:11:31Z
prometheus-node-exporter-63dbp, 2017-02-15T19:11:31Z
prometheus-node-exporter-73bsj, 2017-02-15T19:11:31Z
prometheus-node-exporter-c06lp, 2017-02-15T19:11:31Z
prometheus-node-exporter-j66mz, 2017-02-15T19:11:31Z
default-http-backend-4080621718-hgs3p, 2017-02-15T19:11:31Z
tectonic-console-1860170063-2qhqk, 2017-02-15T19:11:31Z
tectonic-ingress-controller-99581103-v6zw3, 2017-02-15T19:11:31Z
prometheus-node-exporter-bs639, 2017-02-15T19:11:31Z
node-agent-8dvpd, 2017-02-15T19:11:32Z
node-agent-d4qw2, 2017-02-15T19:11:32Z
node-agent-jdm8z, 2017-02-15T19:11:32Z
node-agent-kmv7r, 2017-02-15T19:11:32Z
node-agent-plnvl, 2017-02-15T19:11:32Z
tectonic-channel-operator-2015429748-n0lwx, 2017-02-15T19:11:32Z
kube-version-operator-2694564828-9rq1t, 2017-02-15T19:11:32Z
node-agent-295gl, 2017-02-15T19:11:32Z
tectonic-stats-emitter-832851197-zqhv6, 2017-02-15T19:11:32Z
heapster-v1.2.0-3322913966-zg94j, 2017-02-15T19:11:59Z
tectonic-identity-521760200-qz72d, 2017-02-15T19:30:43Z
kubernetes-bootcamp-472176051-tnn3v, 2017-02-15T23:38:56Z
kubernetes-bootcamp-472176051-0c9dj, 2017-02-15T23:38:56Z
kubernetes-bootcamp-472176051-0pbqh, 2017-02-15T23:38:57Z
kubernetes-bootcamp-472176051-77hs6, 2017-02-15T23:38:58Z
kubernetes-bootcamp-472176051-5sfwj, 2017-02-15T23:38:58Z
kubernetes-bootcamp-472176051-0j68d, 2017-02-15T23:38:59Z
kubernetes-bootcamp-472176051-xx94j, 2017-02-15T23:39:00Z
kubernetes-bootcamp-472176051-hlq4n, 2017-02-15T23:39:01Z
kube-dns-4101612645-2g9vk, 2017-02-16T09:08:27Z
kubernetes-bootcamp-472176051-s7d39, 2017-02-16T09:08:27Z
kubernetes-bootcamp-472176051-r6ns2, 2017-02-16T09:08:27Z
hello-world-2106640509-1bwm9, 2017-02-16T22:11:36Z
hello-world-2106640509-907fc, 2017-02-16T22:11:36Z
hello-world-2106640509-s7z19, 2017-02-16T22:11:36Z